docs/architecture/adr/0025-ts-deprecate-enums.md

```diff
@@ -0,0 +1,196 @@
+  be transmitted to functions.
+- String enums are named types that require a member is used, which is inconsistent with the
+  behavior of numeric enums.
+
+These inconsistencies cause increased complexity from guard statements in the best of cases. In the
+worst cases, their limitations may be unknown, and thus unguarded.
+
+### Detection
+
+TypeScript deprecation can be linted using a fairly short ESLint plugin. The code has [already been
+contributed to main][no-enum-lint] as a suggestion. The same PR adds `FIXME` comments for each team
+to address.
+
+### The Enum-like Pattern
+
+In most cases, enums are unnecessary. A readonly (`as const`) object coupled with a type alias
+avoids both code generation and type inconsistencies.
+
+```ts
+// declare the raw data and reduce repetition with an internal type
+const _CipherType = {
+  Login: 1,
+  SecureNote: 2,
+  Card: 3,
+  Identity: 4,
+  SshKey: 5,
+} as const;
+
+type _CipherType = typeof _CipherType;
+
+// derive the enum-like type from the raw data
+export type CipherType = _CipherType[keyof _CipherType];
+
+// assert that the raw data is of the enum-like type
+export const CipherType: Readonly<{ [K in keyof _CipherType]: CipherType }> =
+  Object.freeze(_CipherType);
+```
+
+This code creates a `type CipherType` that allows arguments and variables to be typed similarly to
+an enum. It also strongly types the `const CiperType` so that direct accesses of its members
+preserve type safety. This ensures that type inference properly limits the accepted values to those
+allowed by `type CipherType`. Without the type assertion, the compiler infers `number` in these
+cases:
+
+```ts
+const s = new Subject(CipherType.Login); // `s` is a `Subject<CipherType>`
+const a = [CipherType.Login, CipherType.Card]; // `a` is an `Array<CipherType>`
+const m = new Map([[CipherType.Login, ""]]); // `m` is a `Map<CipherType, string>`
+```
+
+:::warning
+
+- Types that use enums like [computed property names][computed-property-names] issue a compiler
+  error with this pattern. [This issue is fixed as of TypeScript 5.8][no-member-fields-fixed].
+- Certain objects are more difficult to create with this pattern. This is explored in
+  [Appendix A](#appendix-a-mapped-types-and-enum-likes).
+
+:::
+
+## Considered Options
+
+- **Allow enums, but advise against them** - This is the current state of affairs. With this option,
+  teams **must** address the FIXMEs, but _may_ address them by disabling the lint.
+- **Deprecate enum use** - Allow enums to exist for historic or technical purposes, but prohibit the
+  introduction of new ones. Increase the lint to a "warning" and allow the lint to be disabled.
+- **Eliminate enum use** - Prohibit the introduction of any new enum and replace all enums in the
+  codebase with typescript objects. Increase the lint to an "error" and prohibit disabling of the
+  lint.
+
+## Decision Outcome
+
+Chosen option: **Deprecate enum use**
```

Comment:
issue?: I think the route we took is between the definitions of "Deprecate" and "Eliminate". We made the lint rule throw errors (not warnings) but we also allowed and did disable those errors for existing cases

Reply:
[According to the code](https://github.com/bitwarden/clients/blob/main/libs/eslint/platform/no-enums.mjs#L5-L10), it's presently a suggestion-level lint.